backend/ownerSizeTotalWorker.test.js

```diff
@@ -0,0 +1,288 @@
+    const difference = value - this._ownSize
+    this._ownSize = value
+    this.totalSize += difference
+  }
+
+  get kvSize () {
+    return this._kvSize
+  }
+
+  set kvSize (value) {
+    const difference = value - this._kvSize
+    this._kvSize = value
+    this.ownSize += difference
+  }
+
+  get totalSize () {
+    return this._totalSize
+  }
+
+  set totalSize (value) {
+    const difference = value - this._totalSize
+    this._totalSize = value
+    if (this.ultimateOwner) {
+      this.ultimateOwner.totalSize += difference
+    }
+  }
+
+  get ultimateOwner () {
+    if (!this.owner) return null
+    return this.owner.ultimateOwner || this.owner
+  }
+
+  set ultimateOwner (value) {
+    // empty
+  }
+
+  static async saveIndirectResourcesIndex (resourceID: string) {
+    const ownerID = await sbp('chelonia.db/get', `_private_owner_${resourceID}`)
+    let indirectOwnerID = ownerID
+    while ((indirectOwnerID = await sbp('chelonia.db/get', `_private_owner_${indirectOwnerID}`))) {
+      await appendToIndexFactory(`_private_indirectResources_${indirectOwnerID}`)(resourceID)
+    }
+  }
+
+  static registerBillableEntity = appendToIndexFactory('_private_billable_entities')
+
+  static async saveOwner (ownerID: string, resourceID: string) {
+    // Store the owner for the current resource
+    // Use a queue to check that the owner exists, preventing the creation of
+    // orphaned resources (e.g., because the owner was just deleted)
+    await sbp('chelonia.db/set', `_private_owner_${resourceID}`, ownerID)
+    const resourcesKey = `_private_resources_${ownerID}`
+    // Store the resource in the resource index key
+    // This is done in a queue to handle several simultaneous requests
+    // reading and writing to the same key
+    await appendToIndexFactory(resourcesKey)(resourceID)
+    setTimeout(() => this.saveIndirectResourcesIndex(resourceID), (0, Math.random)() * 500 | 0)
+  }
+
+  async init () {
+    await updateSize(this.id, `_private_size_${this.id}`, this.ownSize)
+    if (this.owner) {
+      await this.constructor.saveOwner(this.owner.id, this.id)
+    } else {
+      await this.constructor.registerBillableEntity(this.id)
+    }
+  }
+
+  async newMessage () {
+    const size = (0, Math.random)() * 9999 | 0
+    this.ownSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+
+  async newKvValue () {
+    const val = (0, Math.random)() * 9999 | 0
+    const sign = (0, Math.random)()
+    const size = (sign < 0.5) ? Math.max(-this.kvSize, -val) : val
+    this.kvSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+}
+
+async function * randomOp (iterations) {
+  const mainContracts: Contract[] = []
+  const allContracts: Contract[] = []
+
+  {
+    const contract = new Contract()
+    console.log('Created initial contract', contract.id)
+    await contract.init()
+    mainContracts.push(contract)
+    allContracts.push(contract)
+    yield
```

Comment:
What is this blank `yield` for? (Could you add a comment?)

Reply:
This and all the other empty `yield` represent a step in the simulation.